builtin/v16/migration/miner.go

```diff
@@ -76,6 +81,67 @@ func (m minerMigrator) MigrateState(ctx context.Context, store cbor.IpldStore, i
 		}
 	}
 
+	// Copy over the deadlines into the new structure, but add the new LivePower and DailyFee fields.
+	// LivePower can be calculated from the partitions, and DailyFee starts at zero for existing
+	// miners.
+
+	deadlines, err := inState.LoadDeadlines(adtStore)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to load deadlines for %s: %w", in.Address, err)
+	}
+	var newDeadlines miner16.Deadlines
+	for dlIdx := range deadlines.Due {
+		oldDeadline, err := deadlines.LoadDeadline(adtStore, uint64(dlIdx))
+		if err != nil {
+			return nil, xerrors.Errorf("failed to load deadline %d for %s: %w", dlIdx, in.Address, err)
+		}
+
+		// Copy old deadline
+		newDeadline := miner16.Deadline{
+			Partitions:                        oldDeadline.Partitions,
+			ExpirationsEpochs:                 oldDeadline.ExpirationsEpochs,
+			PartitionsPoSted:                  oldDeadline.PartitionsPoSted,
+			EarlyTerminations:                 oldDeadline.EarlyTerminations,
+			LiveSectors:                       oldDeadline.LiveSectors,
+			TotalSectors:                      oldDeadline.TotalSectors,
+			FaultyPower:                       miner16.NewPowerPair(oldDeadline.FaultyPower.Raw, oldDeadline.FaultyPower.QA),
```

Comment:
Ew. We really should move power pair into some form of shared type package.

Reply:
Yes, I'm mentally making a list. I did Batch return already but we just need to do all the utility types that don't change.